rotkehlchen/globaldb/updates.py

```diff
@@ -300,74 +305,219 @@ def _parse_full_insert(self, insert_text: str) -> AssetData:
+            if local_asset is None:
+                try:  # if asset is not known then simply do an insertion
+                    with connection.savepoint_ctx() as cursor:
+                        executeall(cursor, full_insert)
+                except sqlite3.Error as e:
+                    self.msg_aggregator.add_warning(
+                        f'Failed to add asset {remote_asset_data.identifier} in the '
+                        f'DB during the v{version} assets update. Skipping entry. '
+                        f'Error: {str(e)}',
+                    )
+                return  # fail or succeed continue to next entry
+
+            if local_asset is not None:
+                AssetResolver().clean_memory_cache(local_asset.identifier.lower())
+
+            # otherwise asset is known, so it's a conflict. Check if we can resolve
+            resolution = conflicts.get(local_asset) if conflicts else None
+            if resolution == 'local':
+                # do nothing, keep local
+                return
+            if resolution == 'remote':
+                try:
+                    with connection.savepoint_ctx() as cursor:
+                        _force_remote(cursor, local_asset, full_insert)
+                except sqlite3.Error as e:
+                    self.msg_aggregator.add_warning(
+                        f'Failed to resolve conflict for {remote_asset_data.identifier} in '
+                        f'the DB during the v{version} assets update. Skipping entry. '
+                        f'Error: {str(e)}',
+                    )
+                return  # fail or succeed continue to next entry
+
+            # else can't resolve. Mark it for the user to resolve.
+            # TODO: When assets refactor is finished, remove the usage of AssetData here
+            local_data = GlobalDBHandler().get_all_asset_data(  # pylint: disable=unsubscriptable-object  # noqa: E501
+                mapping=False,
+                serialized=False,
+                specific_ids=[local_asset.identifier],
+            )[0]
+            self.conflicts.append((local_data, remote_asset_data))
+
     def _apply_single_version_update(
             self,
-            connection: DBConnection,
+            connection: 'DBConnection',
             version: int,
             text: str,
             conflicts: Optional[dict[Asset, Literal['remote', 'local']]],
     ) -> None:
+        """
+        Process each action in the queried data and process it by:
+        1. Try to deserialize it as an asset
+        2. Try to deserialize it as an asset collection
+        3. Try to deserialize it as an asset collection mapping.
+
+        If none of the above rules sucess then the entry is skipped. In the case of assets
+        possible conflicts are handled.
+        """
         lines = text.splitlines()
         for action, full_insert in zip(*[iter(lines)] * 2):
-            if full_insert == '*':
+            if full_insert.replace(' ', '') == '*':
```

Comment:
Why is this needed?

Reply:
While writing the tests I came to a line that required this due to how it was formatted. I believe is not that painful but if not I can change. We take care of this but since we are comparing to `'*'` I believe it makes sense

Reply:
Yeah, I don't see any big problem here. The only thing is that maybe `.strip()` is better to use

Reply:
ah good. I tough strip didn't remove if them it was after. Thanks